src/core/agent/services/signifyNotificationService.ts

```diff
@@ -0,0 +1,56 @@
+import { AgentService } from "./agentService";
+import { GenericRecordType, KeriNotification } from "../agent.types";
+import { Notification } from "./credentialService.types";
+import { NotificationRoute } from "../modules/signify/signifyApi.types";
+export class SignifyNotificationService extends AgentService {
+  async onNotificationKeriStateChanged(
+    callback: (event: KeriNotification) => void
+  ) {
+    // eslint-disable-next-line no-constant-condition
+    while (true) {
+      const notifications = await this.agent.modules.signify.getNotifications();
+      for (const notif of notifications.notes) {
+        await this.processNotification(notif, callback);
+      }
+      await new Promise((rs) => {
+        setTimeout(() => {
+          rs(true);
+        }, 20000);
```

Comment:
Isn't 20 seconds a bit slow to fetch new notifications?

Reply:
@iFergal Is 8 secs ok?

Reply:
It should be very quick to iterate over, right? So can we check every 2 seconds?

Reply:
I will change it to 2 secs